### PR TITLE
[10.0][ADD] website_sale_order_type

### PR DESCRIPTION
**Pull request by SimoRubi**
Bridge module for sale_order_type and website_sale

### CODE REVIEW

File: website_sale_order_type/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    @api.multi
+    def _cart_update(
+            self, product_id=None, line_id=None, add_qty=0, set_qty=0,
+            attributes=None, **kwargs):
+        self.ensure_one()
+        res = super(SaleOrder, self)._cart_update(
+            product_id=product_id, line_id=line_id,
+            add_qty=add_qty, set_qty=set_qty, attributes=attributes, **kwargs)
+        self.onchange_partner_id()
```
Comment (pedrobaeza) [inline website_sale_order_type/models/sale_order.py]: This makes more than only assigning the sale order type. I think you should restrict this or put specifically the code for sale order type handling.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: website_sale_order_type/models/sale_order.py (hunk 1 above)
Comment (SimoRubi) [reply website_sale_order_type/models/sale_order.py]: Thanks for your review, done in https://github.com/OCA/e-commerce/pull/253/commits/8ff2990edeca6e79ca073c626e165fc0dfcff593

**Comment by SimoRubi** [comment]
Depends on https://github.com/OCA/sale-workflow/pull/716

**Comment by SimoRubi** [comment]
Rebased on 10.0, @pedrobaeza could you update your review? Thanks

**Review by pedrobaeza — APPROVED** [review]

**Comment by SimoRubi** [comment]
@OCA/e-commerce-maintainers could you please check? Thanks

**Comment by rafaelbn** [comment]
Hello @SimoRubi ,

I'm going to test it now.

About codecov/patch — 83.33% of diff hit (target 91.56%) I know it is not mandatory but is you think is affordable it could be great! :smile: 

Regards

**Review by rafaelbn — COMMENTED** [review]
Hi @SimoRubi,

Could you please add a USAGE secction?

I'm not sure how to teste this module.

I guess it is not usable for public users and only for registered users?

Thanks!

File: website_sale_order_type/README.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    :target: https://translation.odoo-community.org/projects/e-commerce-10-0/e-commerce-10-0-website_sale_order_type
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/113/10.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+This module allows `sale_order_type <https://github.com/OCA/sale-workflow/tree/10.0/sale_order_type>`__ to work with website_sale.
+
```
Comment (rafaelbn) [inline website_sale_order_type/README.rst:29]: `+This module is useful for users which have definer a sale order type. When a user with a sale order type defined in the Partner form, then the sale order from the ecommerce will be generated with the same type. Remember that the sale order type is defined contact by contact and not in Company level`

File: website_sale_order_type/README.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+**Table of contents**
+
```
Comment (rafaelbn) [inline website_sale_order_type/README.rst:36]: `**Usage**`

+Go in the backend (Sales -> Cursomers) and set to one contact a sale order type.
+As the customer logged in Odoo buy in the ecommerce
+You can check that the sale oder is created with the same sale order type as the contact

**Review by rafaelbn — APPROVED** [review]
Tested in runbot

File: website_sale_order_type/README.rst (hunk 2 above)
Comment (SimoRubi) [reply website_sale_order_type/README.rst:29]: Done in https://github.com/OCA/e-commerce/pull/253/commits/7a995f9da1982d2619c200bb847009ec9b9feb10

**Comment by SimoRubi** [comment]
Big thanks @rafaelbn for your functional review, please check if the changes are good.

I also added the automated tests, I already had something stashed but it wasn't working due to some pesky `env` issues, after [your pep talk](https://github.com/OCA/e-commerce/pull/253#issuecomment-452388993) I decided to go for it and finally found the solution [here](https://odoo-development.readthedocs.io/en/latest/qa/js/phantom_js-test_cr.html).

Now coverage is :heavy_check_mark:

**Pull request merged**